homework_14_140123/src/MethodsImplementation.java

```diff
@@ -0,0 +1,409 @@
+                }
+            } while (i != -1);
+            do {
+                j = bufferedReader2.read();
+                if (j != -1) {
+                    line2 += (char) j;
+                }
+            } while (j != -1);
+            do {
+                k = bufferedReader3.read();
+                if (k != -1) {
+                    line3 += (char) k;
+                }
+            } while (k != -1);
+            if (line1.equals(line2)) {
+                System.out.println("The contents of file 'firstTestFile' and file 'secondTestFile' are the same");
+            } else {
+                System.out.println("The contents of file 'firstTestFile' and file 'secondTestFile' do not match");
+            }
+            if (line1.equals(line3)) {
+                System.out.println("The contents of file 'firstTestFile' and file 'thirdTestFile' are the same");
+            } else {
+                System.out.println("The contents of file 'firstTestFile' and file 'thirdTestFile' do not match");
+            }
+        } catch (IOException e) {
+            System.out.println("File not found");
+        } finally {
+            try {
+                bufferedReader1.close();
+                bufferedReader2.close();
+                bufferedReader3.close();
+            } catch (IOException e) {
+                System.out.println(e);
+                ;
+            }
+        }
+    }
+
+    public static void compareFiles() {
+        File file1 = new File("1.png");
+        File file2 = new File("2.png");
+        File file3 = new File("3.png");
+        BufferedInputStream bufferedInputStream1 = null;
+        BufferedInputStream bufferedInputStream2 = null;
+        BufferedInputStream bufferedInputStream3 = null;
+        try {
+            bufferedInputStream1 = new BufferedInputStream(new FileInputStream(file1));
+            bufferedInputStream2 = new BufferedInputStream(new FileInputStream(file2));
+            bufferedInputStream3 = new BufferedInputStream(new FileInputStream(file3));
+            int i;
+            int j;
+            int k;
+            byte[] arrBytes1 = bufferedInputStream1.readAllBytes();
+            byte[] arrBytes2 = bufferedInputStream2.readAllBytes();
+            byte[] arrBytes3 = bufferedInputStream3.readAllBytes();
+            if (Arrays.equals(arrBytes1, arrBytes2)) {
+                System.out.println("The contents of file '1' and file '2' are the same");
+            } else {
+                System.out.println("The contents of file '1' and file '2' do not match");
+            }
+            if (Arrays.equals(arrBytes1, arrBytes3)) {
+                System.out.println("The contents of file '1' and file '3' are the same");
+            } else {
+                System.out.println("The contents of file '1' and file '3' do not match");
+            }
+        } catch (IOException e) {
+            System.out.println("File is not found");
+        } finally {
+            try {
```

Comment:
you don't need try block here, just close stream

Reply:
fixed. using try with resources.